Code/Framework/AzToolsFramework/AzToolsFramework/UI/UICore/TargetSelectorButton.cpp

```diff
@@ -121,46 +131,49 @@ namespace AzToolsFramework
 
     void TargetSelectorButton::UpdateStatus()
     {
-        AzFramework::TargetInfo info;
-        EBUS_EVENT_RESULT(info, AzFramework::TargetManager::Bus, GetDesiredTarget);
-        if (!info.GetPersistentId())
+        auto* remoteToolsInterface = AzFramework::RemoteToolsInterface::Get();
+        if (remoteToolsInterface)
         {
-            this->setIcon(QIcon(":/general/target_none"));
-            this->setText("Target: None");
-            return;
-        }
+            AzFramework::RemoteToolsEndpointInfo info = AzFramework::RemoteToolsInterface::Get()->GetDesiredEndpoint(m_remoteToolsKey);
+            if (!info.GetPersistentId())
+            {
+                this->setIcon(QIcon(":/general/target_none"));
+                this->setText("Target: None");
+                return;
+            }
 
-        if (!info.IsValid())
-        {
-            this->setIcon(QIcon(":/general/target_none"));
-            this->setText("Target: None");
-            return;
-        }
+            if (!info.IsValid())
+            {
+                this->setIcon(QIcon(":/general/target_none"));
+                this->setText("Target: None");
+                return;
+            }
 
-        this->setText(QString("Target: %1").arg(info.GetDisplayName()));
-        if (info.GetStatusFlags() & AzFramework::TF_ONLINE)
-        {
-            this->setIcon(QIcon(":/general/target_connected"));
-        }
-        else
-        {
-            this->setIcon(QIcon(":/general/target_disconnected"));
+            this->setText(QString("Target: %1").arg(info.GetDisplayName()));
+
+            if (AzFramework::RemoteToolsInterface::Get()->IsEndpointOnline(m_remoteToolsKey, info.GetNetworkId()))
```

Comment:
This is code is repeating grabbing the RemoteToolsInterface via the AZ::Interface, when it already has it stored in the local `remoteToolsInterface` variable.